en/lessons/basics/operators.md

```diff
@@ -0,0 +1,185 @@
+ghci> 2010 `mod` 1998
+12
+ghci> (`div` 3) . (* 1471) $ 5
+2451
+```
+
+
+### Operator Precedence and Associativity
+
+An operator's precedence and associativity combined is called its fixity, and it
+controls how the operator is parsed in an expression. To find out the fixity of an
+operator, look through `ghci`'s `:info` or the documentation for the operator:
+if fixity information isn't displayed, it defaults to `infixl 9`. The same applies to
+regular functions used in infix mode.
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> :info (+)
+infixl 6 +
+ghci> :i ($)
+infixr 0 $
+ghci> :i (==)
+infix 4 ==
+ghci> :i elem
+infix 4 `elem`
+```
+
+In Haskell, operators have a precedence level between 0 and 9: this precedence level
+is the number in the operator's fixity. To see how a group of expressions
+separated by operators would be parenthesised, surround the highest precedence operator
+and the expressions next to it in parentheses, and repeat this process until there
+is only one operator left at the highest level of the expression. Here's an example of
+applying this process:
+```
+16425973 + 17149685 * (106 + 235) == 3530784511
+         |          │              ╰─ Precedence level 4
+         |          ╰─ Precedence level 7
+         ╰─ Precedence level 6
+16425973 + (17149685 * (106 + 235)) == 3530784511
+(16425973 + (17149685 * (106 + 235))) == 3530784511
+```
+
+Knowing operator precedence is helpful in writing code without unnecessary parentheses:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> (^ (3 - 1)) . subtract 3818 $ 4
+14546596
+ghci> (1729 + 1) * 1 `div` 2
+865
+ghci> 1 + 6 <= 7952317874 + 6 && 0 == 0
+True
+```
+
+However, this process may be ambiguous: multiple operators of the highest precedence
+could exist. Operator associativity solves this problem. An operator has either left
+(`infixl`), right (`infixr`), or neutral (`infix`) associativity. If either all the
+operators of the highest precedence in an expression have left associativity, or they
+all have right associativity, you can parenthesise the expression by selecting
+the leftmost or rightmost operator of highest precedence accordingly. For example,
+given that both `+` and `-` have left associativity, you could do the following
+process to parenthesise an expression:
+```
+196418 - 514229 + 317811 - 5605
+(196418 - 514229) + 317811 - 5605
+((196418 - 514229) + 317811) - 5605
+```
+
+Otherwise, if multiple operators of the highest precedence exist, and either one has
+neutral associativity or they differ in associativity, Haskell will throw an error:
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> 29 + 61 == 3 == True
+<interactive>:1:1: error:
+    Precedence parsing error
+        cannot mix ‘==’ [infix 4] and ‘==’ [infix 4] in the same infix expression
+ghci> 0 <> 1 + (-1)
+<interactive>:2:1: error:
+    Precedence parsing error
+        cannot mix ‘<>’ [infixr 6] and ‘+’ [infixl 6] in the same infix expression
```

Comment:
nice example

Reply:
Thanks. It's a small easter egg because it can typecheck if you bracket it: `(0 <> 1) + (-1) :: Sum Integer`.